fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -1571,4 +1574,14 @@ public <T> CompletableFuture<T> doWithReadLock(@Nonnull final LockIdentifier ide
     public <T> CompletableFuture<T> doWithWriteLock(@Nonnull final LockIdentifier identifier, @Nonnull final Supplier<CompletableFuture<T>> operation) {
         return lockRegistry.doWithWriteLock(identifier, operation);
     }
+
+    /**
+     * Returns a transaction-bound factory of {@link TempTable}s.
+     * @return a factory of {@link TempTable}s bound to this transactional context.
+     */
+    @API(API.Status.INTERNAL)
+    @Nonnull
+    public TempTable.Factory getTempTableFactory() {
```

Comment:
The `FDBRecordContext` should be creating temp tables and handing out objects on request. It should keep track of what has been handed out so when the context closed, the temp tables can be cleaned up (if they touch disk).

Reply:
That's too much responsibility of a context (which is supposed to be wrapper of FDB transactions I think), therefore, I created a factory to achieve three goals:
1. centralize where temp table are created and handed out, so it can do accounting, life-cycle, ... etc.
2. separation of concerns.
3. injection for testing, e.g. I want to be able to test requests for temp tables by the execution without necessarily having to inherit / modify the `FDBTransactionContext`.